### PR TITLE
[TEST]: 5303: Y-Flow: Reroute

### PR DESCRIPTION
**Pull request by yuliiamir**
Y-Flow reroute suite has been expanded due to the issue https://github.com/telstra/open-kilda/issues/5303.

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy [hunk 1]
```diff
@@ -141,4 +150,197 @@ class YFlowRerouteSpec extends HealthCheckSpecification {
...
+            [islUtils.toLinkProps(isl, ["cost": (0).toString()])]
+        })
+
+        and: "Y-Flow with shared path has been created successfully"
+        def yFlow = yFlowHelper.addYFlow(yFlowHelper.randomYFlow(swT, false))
+        def yFlowPathBeforeReroute = northboundV2.getYFlowPaths(yFlow.YFlowId)
+
+        and: "Shared ISLs cost has been changed to provide on-demand Y-Flow reroute"
+        def sharedPathIslBeforeReroute = new Path(yFlowPathBeforeReroute.sharedPath.forward + yFlowPathBeforeReroute.sharedPath.reverse, topology).getInvolvedIsls()
+        northbound.updateLinkProps(sharedPathIslBeforeReroute.collectMany { isl ->
```
Comment (pkazlenka) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy:207]: Can we move this to some helper or IslExtended class?

**Review by pkazlenka — CHANGES_REQUESTED** [review]

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy [hunk 2]
```diff
@@ -141,4 +150,197 @@ class YFlowRerouteSpec extends HealthCheckSpecification {
...
+        def yFlow = yFlowHelper.addYFlow(yFlowRequest)
+        def yFlowPathBeforeReroute = northboundV2.getYFlowPaths(yFlow.YFlowId)
+
+        when: "Y-Flow reroute has been called"
+        northboundV2.rerouteYFlow(yFlow.YFlowId)
+
+        then: "The appropriate error has been returned"
+        def actualException = thrown(HttpClientErrorException)
+        new YFlowRerouteExpectedError(~/Reroute is unsuccessful. Couldn't find new path\(s\)/).matches(actualException)
+        actualException.message.contains("The flow already on the best path")
```
[the remark this answers is not in the document]
Comment (yuliiamir) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy:169]: In the scope of error checking we verify 
```
status-code
error-message
error-description
```
But this part (`The flow already on the best path`) has been specified as `error-type` and it is a part of the error message on GUI.

FlowIsOnTheBestPath

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy (hunk 1 above)
Comment (yuliiamir) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy]: Done.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy (hunk 2 above)
[the remark this answers is not in the document]
Comment (pkazlenka) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy:169]: I'd say we should enforce developers to follow a convention on error message fields.

[the remark this answers is not in the document]
Comment (yuliiamir) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowRerouteSpec.groovy]: Discussed, the appropriate changes have been added.

**Review by pkazlenka — APPROVED** [review]

**Review by IvanChupin — APPROVED** [review]

**Pull request merged**